### ISSUE TITLE
BeepManager plays two beeps sometimes

### ISSUE THREAD

**Issue by mjmfighter** [journeyapps/zxing-android-embedded#155]
I create a customized continuous capture activity and am using the beep manager to play a sound when a new barcode is scanned.  However, sometimes the beep manager plays the beep sound more than once even though it gets called only once.

Here is what I have in my BarcodeCallback.barcodeResult function:

```
if(Config.getInstance().addBarCode(result.getText())) {  //synchronized method!
    Log.i("MjMTest", "Played sound for: " + result.getText());
    beepManager.playBeepSoundAndVibrate();
}
```

As seen, the barcode is added to a config instance.  This adds it to a synchronized list to make sure it doesn't get added more than once if it has already scanned.  It then prints to the log that it has played a sound for X barcode, then plays the sound.  In my logs, I will only get the output for the log printed once, but the beep sound will still play twice.  The two beeps usually happen on the first barcode I can, but sometimes happen on other barcodes after the first scan as well.

Is there anything I am missing?  Or is the beep manager a bit buggy like this sometimes?  Thanks for any help

**Comment by Bajranghudda1** [comment]
@mjmfighter can you reproduce it??
I am also using BeepManager but i didn't see any problem, there might be your barcode scanned two times frequently, you can have a check if barcode scanned first time then stop immediately scanning.

**Comment by mjmfighter** [comment]
@Bajranghudda1 I reproduce it every time.

And as said above, the barcode can and possibly does scan more than once... however the sound is only ever played once.  I first check the barcode against a synchronized list to see if its been scanned or not.  If it has not been scanned this function adds it to the list and returns true, or else it returns false.  If it returns true it then uses the above function and prints that log and then plays the sound.  In my debug log, the "Played sound for" log is only printed once, but I still hear two short beeps being played.

**Comment by roxma** [comment]
same issue here

**Comment by roxma** [comment]
Link to [this PR](https://github.com/journeyapps/zxing-android-embedded/pull/221)

**Comment by areebbeigh** [comment]
This is obvious, when you create a continuous scanner activity the camera continuously scans for barcodes so each time a barcode is scanned the callback method is executed. A work around to this is to call the pause() method on the barcode view each time a scan is done and in the onResume() overridden method of your activity call resume() method on the barcode view.

**Comment by mjmfighter** [comment]
@areebbeigh This is not obvious.  Please re-read my first post.  Yes I am using a continuous scanner, but I am only ever calling the  bm.playBeepSoundAndVibrate() once.  I check all the barcodes sent through that method and even log some output when the beep manager is called.  It only ever prints the log once, therefore I know it is only being called once.

**Comment by areebbeigh** [comment]
@mjmfighter Sorry, my bad.

**Comment by roxma** [comment]
I’m using the continuous scanner too.
I believe that the mediaplayer.seek call in the beep manager triggers multipln beeps somehow. So I sent a PR to fix this. It works for me.

**Comment by rkistner** [comment]
This should be fixed with #221 now - thanks @roxma.
